test/gtest/common/test.cc

```diff
@@ -141,9 +143,9 @@ test_base::count_warns_logger(const char *file, unsigned line, const char *funct
 std::string test_base::format_message(const char *message, va_list ap)
 {
     const size_t buffer_size = ucs_log_get_buffer_size();
-    char buf[buffer_size];
-    vsnprintf(buf, buffer_size, message, ap);
-    return std::string(buf);
+    std::vector<char> buf(buffer_size);
```

Comment:
std::string

Reply:
bad idea: std::string prohibits direct write into buffer

Reply:
i'm pretty sure you can do &buf[0] on std::string just as for std::vector..

Reply:
ok, done

Reply:
doesn't work - tests failed